### ISSUE TITLE
Bermudan Callable Bonds

### ISSUE THREAD

**Issue by noonediesalone** [OpenSourceRisk/Engine#162]
Hi,

Is there an easy way to represent a bermudan style callable bond in ORE v.10?
A callable swap is possible as a deal with a non-call swap and a swaption with bermudan style.
Also a vanilla bond is possible and an european bond option, but I can't straight away find a combo that captures the bermudan embedded option.

Regards,
Laurentiu.

**Comment by pcaspers** [comment]
Hi Laurentiu,

there is no direct way at the moment. Callable / Puttable Bonds are under development and will most probably added in one of the next releases. This will include American-style calls.

In some contexts callable bonds are modelled as a vanilla bond + interest rate swaption exercising into an asset swap for the bond. This is of course already possible.

Best
Peter

**Comment by noonediesalone** [comment]
Hi Peter,

Thank you for your comments. I've raised this issue with hopes this will get added with one of the next releases, so glad to hear you're confirming this 😄 I think will let this issue open for a little while longer.
Having the embedded option as a swaption for an asset swap is a plausible way to go. Maybe a drawback is that swap and bond will not be discounted by the same thing (e.g. asset swap can carry collateral, while bond needs some benchmark yield to discount).

Regards,
Laurentiu,

**Comment by pcaspers** [comment]
Absolutely, it's not the same thing! The spread on the asset swap's funding leg takes the role of the discounting spread but it's clearly not equivalent and can give very different answers.
